net: shell: Add support to set Ethernet MAC address

krish2718 opened the pull request:
Add a shell command to set Ethernet MAC address, useful in configuring custom/local MAC addresses.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>

krish2718 commented:
@sachinthegreen @sr1dh4r FYI

jukkar commented on subsys/net/ip/net_shell.c:
```diff
@@ -3563,6 +3566,51 @@ static int cmd_net_iface(const struct shell *shell, size_t argc, char *argv[])
+		return -ENOEXEC;
+	}
+
+	ret = net_if_set_link_addr(iface, mac_addr.addr, sizeof(mac_addr), NET_LINK_ETHERNET);
```
The system will disallow changing of the MAC address if the interface is running. Perhaps we could check the error code here (-EPERM) and print a better message if that happens.

jukkar reviewed (dismissed):
LGTM

krish2718 commented:
* Added a new commit to fix duplicate tag (struct shell *shell) everywhere
* Fixed few issues: Defining ret, missing command in command definition
* Used `goto err` instead of returning from all negative paths

rlubos commented:
Something's off with [97ac193](https://github.com/zephyrproject-rtos/zephyr/pull/53539/commits/97ac1938157c89c3c195e8ebfaad816ebc6f1175), the git commit author name is lowercase (`krishna T`), it seems to make CI unhappy.

Commit:
net: shell: Fix duplicate tag

Recently introduced MISRA-C CI check complains about use of a duplicte
tag, so, shorten the variable from shell->sh to avoid same name as the
structure.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>

Commit:
net: shell: Add support to set Ethernet MAC address

Add a shell command to set Ethernet MAC address, useful in configuring
custom/local MAC addresses.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>
